src/raftstore/coprocessor/mod.rs

```diff
@@ -72,19 +71,15 @@ pub trait QueryObserver: Coprocessor {
     /// Hook to call before proposing write request.
     ///
     /// We don't propose read request, hence there is no hook for it yet.
-    fn pre_propose_query(
```

Comment:
I would suggest to keep the requests parameters. How about changing it to `&mut Vec<Request>`?

Reply:
+1, I think we still need to keep the parameter.

Reply:
Why do you want to keep them? They are not used anywhere that I can see. If we might need them in the future, then we can add them back in then.

Reply:
if you want to remove them, you can use another PR, but not here. 

IMO, this PR only introduces `into`, doesn't do other things.

Reply:
Actually, it's a public API. All those are designed for community usages, people can add customizing functionality to TiKV via those APIs. You can consider them as plugins.

Allow changing the requests provides more flexibility. For example, there is a potential optimization to reduce the write batch size by pushing down transaction logic in such APIs.

Reply:
:+1: